Sources/MongoSwift/MongoClient.swift

```diff
@@ -304,6 +304,104 @@ public class MongoClient {
+     }
+
+     /**
+      * Starts a `ChangeStream` on a `MongoClient`. Allows the client to observe all changes in a cluster - excluding
+      * system collections i.g. "config", "local", and "admin" databases. Associates the specified `Codable` type `T`
+      * with the `fullDocument` field in the `ChangeStreamDocument` emitted by the returned `ChangeStream`.
+      * - Parameters:
+      *   - pipeline: An array of aggregation pipeline stages to apply to the events returned by the change stream.
+      *   - options: An optional `ChangeStreamOptions` to use when constructing the change stream.
+      *   - session: An optional `ClientSession` to use with this change stream.
+      *   - withFullDocumentType: The type that the change events emitted from the change stream will be decoded to.
+      * - Returns: A `ChangeStream` on all collections in all databases in a cluster.
+      * - Throws:
+      *   - `ServerError.commandError` if an error occurs on the server while creating the change stream.
+      *   - `ServerError.commandError` if the pipeline passed is invalid.
+      *   - `UserError.invalidArgumentError` if the options passed formed an invalid combination.
+      *   - `UserError.invalidArgumentError` if the `_id` field is projected out of the change stream documents by the
+      *     pipeline.
+      * - SeeAlso:
+      *   - https://docs.mongodb.com/manual/changeStreams/
+      *   - https://docs.mongodb.com/manual/meta/aggregation-quick-reference/
+      *   - https://docs.mongodb.com/manual/reference/system-collections/
+      * - Note: Supported in MongoDB version 4.0+ only.
+      */
+     public func watch<T: Codable>(_  pipeline: [Document] = [],
+                                   options: ChangeStreamOptions?  =  nil,
+                                   session: ClientSession? = nil,
+                                   withFullDocumentType: T.Type) throws ->
+                                   ChangeStream<ChangeStreamDocument<T>> {
+        return try self.watch(pipeline,
+                              options: options,
+                              session: session,
+                              withEventType: ChangeStreamDocument<T>.self)
+     }
+
+     /**
+      * Starts a `ChangeStream` on a `MongoClient`. Allows the client to observe all changes in a cluster - excluding
+      * system collections i.g. "config", "local", and "admin" databases. Associates the specified `Codable` type `T`
+      * with the returned `ChangeStream`.
+      * - Parameters:
+      *   - pipeline: An array of aggregation pipeline stages to apply to the events returned by the change stream.
+      *   - options: An optional `ChangeStreamOptions` to use when constructing the change stream.
+      *   - session: An optional `ClientSession` to use with this change stream.
+      *   - withEventType: The type that the entire change stream response will be decoded to.
```

Comment:
maybe add on to the end of this something like "and that will be returned when iterating through the change stream" to make it clear that changestream.next() returns this type